crates/katana/core/src/service/messaging/starknet.rs

```diff
@@ -103,8 +99,11 @@ impl StarknetMessaging {
                 break;
             }
         }
-
-        Ok(block_to_events)
+    
+        // Ordering the events to process them in the correct order
+        events.sort_by_key(|e| e.block_number);
```

Comment:
```suggestion
```
When we use `get_events`, blocks are already ordered (and events are too). What made the order flaky, it's because before we were inserting into a `HashMap`.

So in this case, just using the `events.push(event)`, the order is guaranteed, as we're processing the blocks as they are gathered from `get_events`.

So this extra ordering is not required.